src/junit.rs

```diff
@@ -195,6 +248,14 @@ pub fn parse_junit_xml(file_bytes: &[u8]) -> PyResult<ParsingInfo> {
                     testrun.outcome = Outcome::Failure;
 
                     testrun.failure_message = get_attribute(&e, "message")?;
+
+                    if let Some(failure_message) = &testrun.failure_message {
+                        testrun.failure_message = Some(
+                            unescape(&failure_message)
+                                .unwrap_or(Cow::Borrowed(testrun.failure_message.as_ref().unwrap()))
+                                .to_string(),
+                        );
+                    }
```

Comment:
```suggestion
testrun.failure_message = testrun.failure_message.map(|failure_message| {
unescape(&failure_message).map(Cow::into_owned).unwrap_or(failure_message)
})
```

I believe this should work, though I’m unsure whether this might throw a borrowchecker error.

… or better yet, since this code is duplicated, and you are setting `failure_message` right before. You can just move this to a function with signature `&str -> String` which enforced the unescaping. Then you can just `Option::map` that wherever you need.